### PR TITLE
Alarm when batch jobs fails

### PR DESCRIPTION
**Pull request by eidelmanjonathan**
This patch has us generate a lambdaAlarm if we ever have a batch job fail

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ci/snapshot/cbmc_ci_end.py [hunk 1]
```diff
@@ -147,7 +147,10 @@ def lambda_handler(event, context):
     job_id = event["detail"]["jobId"]
     status = event["detail"]["status"]
     job_name_info = Job_name_info(job_name)
-    if (status in ["SUCCEEDED", "FAILED"] and
```
Comment (markrtuttle) [inline ci/snapshot/cbmc_ci_end.py:150]: This appears to raise an alarm in the event of every failure of every proof, and not just the existence of a failure on a pull request.  Also, we certainly don't want to be alarming on proof failures in draft pull requests.

File: ci/snapshot/cbmc_ci_end.py [hunk 2]
```diff
@@ -147,7 +147,10 @@ def lambda_handler(event, context):
...
+    if status in ["FAILED"]:
+        print(f"ERROR: The following job has failed: {job_name} with status {status}")
+        raise Exception(f"The following job has failed: {job_name} with status {status}")
```
Comment (markrtuttle) [inline ci/snapshot/cbmc_ci_end.py:152]: This f-string stuff is not going to work in CI that is running Ubuntu16 and Python 3.5 (the f-string stuff is 3.7 or 3.8)

File: ci/snapshot/cbmc_ci_end.py (hunk 1 above)
Comment (eidelmanjonathan) [reply ci/snapshot/cbmc_ci_end.py:150]: This fails any time any AWS Batch job has a failure status. Even if proofs fail, the batch job should terminate successfully. If it doesn't, this is a CI bug. The downside here is that if the build job fails all other jobs will fail and we will have many alarms, but I don't see a way around that

File: ci/snapshot/cbmc_ci_end.py (hunk 2 above)
Comment (eidelmanjonathan) [reply ci/snapshot/cbmc_ci_end.py:152]: This is running in lambda which uses python3.6 (which has f strings)

**Pull request merged**